### PR TITLE
change default deploytype to avoid defaulting to invalid type

### PR DESCRIPTION
**Pull request by davidgamero**
# Description

The default for the deploytype flag is currently set to `.`, which isn't a valid deployment type, and causes a nil pointer error.

In order to catch this in the future, a new smoke test action was added that uses the default for every option.
This smoke test fails with invalid defaults like the one this PR fixes, and we should consider adding one for other commands once we have the ability to extract sane defaults for all of the flags on other commands.

Since the APPNAME variable didn't have a default, I added one to the draft.yaml allowing this smoke test to execute, and providing a basic default for testing.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Validated with new error return on deployment type lookup
Added new smoke test for create command

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by OliverMKing — APPROVED** [review]

**Review by bfoley13 — APPROVED** [review]
Looks good for the bug fix. Lets also think about adding the validation for other defaults where it makes sense

**Review by OliverMKing — APPROVED** [review]

File: template/deployments/helm/draft.yaml [hunk 1]
```diff
@@ -23,3 +23,5 @@ variableDefaults:
   - name: "IMAGETAG"
     value: "latest"
     disablePrompt: true
+  - name: "APPNAME"
```
Comment (OliverMKing) [inline template/deployments/helm/draft.yaml:26]: why is this addition needed?

**Review by OliverMKing — APPROVED** [review]

File: template/deployments/helm/draft.yaml (hunk 1 above)
Comment (davidgamero) [reply template/deployments/helm/draft.yaml:26]: without a default appname, we can't add a test case for taking every default to validate that they work, but it looks like the GH runners don't support the `yes` command anyway, so i may remove the test

Comment (davidgamero) [reply template/deployments/helm/draft.yaml]: good point i'll remove this and defer to a later PR

**Review by bfoley13 — APPROVED** [review]
nice updates for handling the defaults
